### DISCUSSION TITLE
Handling errors thrown from +page.svelte

### DISCUSSION THREAD

**Discussion by pboguslawski** [sveltejs/kit#9428] [Q&A] [answered]
[Manual](https://kit.svelte.dev/docs/errors#responses) explains how to throw an error from load in `+page.ts` (SPA scenario) and render error with `+error.svelte`.

Is it possible to handle errors in same way from JS in `+page.svelte` (i.e. when `+page.svelte` with list of items is rendered, user presses del button which calls async function defined in `<script>` and promise fails `+error.svelte` should be rendered with error)?

With
```
<Button
			on:click={() => {
				deleteItems().catch((error) => {					
					throw error;
				});
			}}
		>
```
`+error.svelte` is not rendered and console shows `Uncaught (in promise)` error.

**Comment by teemingc** [comment, marked as answer]
You can try calling `goto()` to the current page but with a `URLSearchParam` that is checked in the `load` function, then throw an error.

Otherwise, a more convenient method could be introduced to achieve this.

**Comment by pboguslawski** [comment]
Thank you. Passing data from `+page.svelte` to `+page.ts` using store sounds better than `URLSearchParam` (i.e. for larger data). Please let me know it there is better way to pass stuff in this direction than store/URL params.

Consider converting this discussion to issue to add standard way to handle throws from `+page.svelte` with `+error.svelte` (same as in `+page.ts`) and updating manual with this standard solution.
